dot/state/epoch.go

```diff
@@ -68,7 +68,7 @@ type EpochState struct {
 }
 
 // NewEpochStateFromGenesis returns a new EpochState given information for the first epoch, fetched from the runtime
-func NewEpochStateFromGenesis(db chaindb.Database, blockState *BlockState,
+func NewEpochStateFromGenesis(db *chaindb.BadgerDB, blockState *BlockState,
```

Comment:
This is left as an implementation on purpose, since it should be changed by issue #2981